editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -520,10 +520,13 @@ export const previewStore: StateCreator<
 
     // Only proceed if the user has seen at least one node with this fn before
     const visitedFns = Object.entries(breadcrumbs).filter(
-      ([nodeId, _breadcrumb]) =>
-        flow[nodeId].data?.fn === data.fn ||
+      ([nodeId, _breadcrumb]) =>{
+        if (!flow[nodeId]) {
+          return;
+        }
+        return flow[nodeId].data?.fn === data.fn ||
```

Comment:
When written this way, it makes it impossible for the `||` clause to be hit (which does not rely on the breadcrumb node still existing in the flow). 

Would a 1-line fix like `flow[nodeId]?.data?.fn === data.fn ||` accomplish the same while still allowing the second "or" clause to be hit?